src/main.ts

```diff
@@ -0,0 +1,124 @@
+    ].filter(el => el != '');
+}
+
+function getGitHubToken(): string {
+    const token = process.env['GITHUB_TOKEN'] || rustCore.input.getInput('github-token');
+    if (!token)
+        throw new Error('Querying the GitHub API is possible only if the GitHub token is set.');
+    return token;
+}
+
+async function getCargoSemverChecksDownloadURL(target: string): Promise<string> {
+    const octokit = github.getOctokit(getGitHubToken());
+
+    const getReleaseUrl = await octokit.rest.repos.getLatestRelease({
+        owner: 'obi1kenobi',
+        repo: 'cargo-semver-checks'
+    });
+
+    const asset = getReleaseUrl.data.assets.find(asset => {
+        return asset['name'].endsWith(`${target}.tar.gz`)
+    });
+
+    if (!asset) {
+        throw new Error(`Couldn't find a release for target ${target}.`);
+    }
+
+    return asset.url;
+}
+
+async function installRustUp(): Promise<void> {
+    const rustup = await rustCore.RustUp.getOrInstall();
+    await rustup.call(['show']);
+    await rustup.setProfile('minimal');
+    await rustup.installToolchain('stable');
+}
+
+async function runCargoSemverChecks(cargo: rustCore.Cargo): Promise<void> {
+    await cargo.call(['semver-checks', 'check-release'].concat(getCheckReleaseArguments()));
+}
+
+async function installCargoSemverChecksFromPrecompiledBinary(): Promise<void> {
+    const url = await getCargoSemverChecksDownloadURL(getPlatformMatchingTarget());
+
+    core.info(`downloading cargo-semver-checks from ${url}`);
+    const tarballPath = await toolCache.downloadTool(url, undefined, `token ${getGitHubToken()}`, {
+        accept: 'application/octet-stream'
+    });
+    core.info(`extracting ${tarballPath}`);
+    const binPath = await toolCache.extractTar(tarballPath, undefined, ["xz"]);
+
+    core.addPath(binPath);
+}
+
+async function installCargoSemverChecksUsingCargo(cargo: rustCore.Cargo): Promise<void> {
+    await cargo.call(['install', 'cargo-semver-checks', '--locked']);
+}
+
+async function installCargoSemverChecks(cargo: rustCore.Cargo): Promise<void> {
+    if (await io.which('cargo-semver-checks') != '') {
+        return;
+    }
+    
+    core.info('cargo-semver-checks is not installed, installing now...');
+
+    try {
+        await installCargoSemverChecksFromPrecompiledBinary();
+    } catch (error: any) {
+        core.info('Failed to download precompiled binary of cargo-semver-checks.');
+        core.info(`Error: ${error.message}`);
+        core.info('Installing using cargo install...');
+
+        await installCargoSemverChecksUsingCargo(cargo);
+    }
+}
+
+async function run(): Promise<void> {
+    await installRustUp();
+
+    const cargo = await rustCore.Cargo.get();
+
+    await installCargoSemverChecks(cargo);
+    await runCargoSemverChecks(cargo);
+}
+
+async function main() {
+    try {
+        await run();
+    } catch (error: any) {
+        core.setFailed(error.message);
+    }
+}
+
+main();
```

Comment:
```suggestion
main();

```

Reply:
Also, should this `await` the call? I'm not up to date on my TypeScript semantics here.

Reply:
Great question! I had to check in order to make myself sure. The promise is executed even if it is not awaited. And in fact, at the top level of a script the `await` is even not allowed.

Reply:
Ugh. I can't believe top-level `await` is still not a supported feature, it's been in the works for ages.